pulsar-client/src/main/java/org/apache/pulsar/client/impl/ProducerImpl.java

```diff
@@ -2209,5 +2234,11 @@ boolean isErrorStat() {
         return errorState;
     }
 
+    @VisibleForTesting
+    CompletableFuture<Void> getOriginalLastSendFuture() {
```

Comment:
Since we added this to both `ProducerImpl` and `PartitionedProducerImpl`, should we consider adding this to the `Producer` interface?

Reply:
Thanks for review. I have addressed your comments. PTAL.